lib/Parse/ParseDecl.cpp

```diff
@@ -8980,6 +8980,23 @@ Parser::parseDeclEnumCase(ParseDeclOptions Flags,
       if (ArgParams.isNull() || ArgParams.hasCodeCompletion())
         return ParserStatus(ArgParams);
     }
+    
+    // See if there are generic params.
+    auto genericResults = maybeParseGenericParams()
+      .getPtrOrNull();
+    if (genericResults && !genericResults->getParams().empty()) {
+      auto genericLoc = genericResults->getParams().front()->getStartLoc();
+      diagnose(genericLoc, diag::generic_param_cant_be_used_in_enum_case_decl);
+      
+      // check if there's a following argument type after
+      // generic declaration and parse it to ignore it.
+      if (Tok.isFollowingLParen()) {
```

Comment:
Because this diagnosis falls through without returning, you can move this generic result check to be done prior to the previous if statement (which parses the argument type)

Reply:
good suggestion. done.